include/rocksdb/cache.h

```diff
@@ -209,12 +213,12 @@ class Cache {
   using CreateCallback = std::function<Status(void* buf, size_t size,
                                               void** out_obj, size_t* charge)>;
 
-  Cache(std::shared_ptr<MemoryAllocator> allocator = nullptr)
```

Comment:
Move the memory_allocator out of the abstract class "Cache" and use it as a member of cache option looks good to me. But one thing I'm not sure is, if it will make some interface compatibility issues for some users how developed their own cache?

Reply:
It goes both ways.  If there is a non-standard Cache that was expecting/relying on the memory allocator, then this will change that class.  OTOH, if there is a cache that does not use the RocksDB allocators (like potentially CacheLib), the implementation would not have to provide one.